api/container.gql

```diff
@@ -22,17 +22,28 @@ Examples:
 scalar ContainerAddress
 
 """
-"An OCI-compatible container, also known as a "docker container"
+An OCI-compatible container, also known as a "docker container"
 """
 type Container {
     "A unique identifier for this container"
     id: ContainerID!
 
-    "Initialize this container from the base image published at the given address"
+    "Initialize this container to the base image published at the given address"
     from(address: ContainerAddress!): Container!
 
+    """
+    Initialize this container to the result of the specified build.
+
+    The build can be customized to use a Dockerfile at a non-standard location,
+    or (in the future) to support files other than Dockerfile.
+    """
+    build(source: DirectoryID!, file: String="./Dockerfile")
+
     "This container's root filesystem. Mounts are not included."
-    rootfs: Directory!
```

Comment:
nit: `root` is a bit confusing with `user` below.

How about just `fs` / `withFS`? `root` is ambiguous between the user and `/root`, `fs` IMO less so

Reply:
Sure. I avoided `fs` because of the capitalization issue (felt weird having `fs` lowercase in one field and uppercase in the other). But I like that it's short. If nobody is weirded out by the capitalization change, I will change to your suggestion.

Reply:
Done.